lib/sequenceserver/blast/error.rb

```diff
@@ -0,0 +1,76 @@
+require 'English'
+
+require_relative 'constants'
+require_relative 'exceptions'
+
+module SequenceServer
+
+  #BLAST module
+  module BLAST
+    class Error
+      def initialize(job)
+        @job = job
+        callError
+      end
+
+      attr_reader :job
+
+      def callError
+        case job.exitStatus
+        when 1 # error in query sequence or options; see [1]
+          validate_options job.advanced_params # if advanced_params are incorrect
+          empty_rfile_error # if rfile is empty
+        when 2, 3, 4, 255
+          fail ArgumentError, readError
```

Comment:
How do you conclude this is ArgumentError? Especially in the light that original code called this scenario RuntimeError?